core/internal/stream/handler.go

```diff
@@ -471,10 +471,13 @@ func (h *Handler) handleRequestRunStart(record *spb.Record, request *spb.RunStar
 	var ok bool
 	run := request.Run
 
-	// offsset by run.Runtime to account for potential run branching
-	startTime := run.StartTime.AsTime().Add(time.Duration(-run.Runtime) * time.Second)
-	// start the run timer
-	h.runTimer.Start(&startTime)
+	// offset the timer by the runtime of the run, this is to account for the
+	// potential run branching
```

Comment:
grammar and punctuation; this can be a lot simpler

```go
// Add on to the previous run time for branched runs.
```

Reply:
will update